### PR TITLE
ENH: Add descriptive comments to spectra filter files

### PR DESCRIPTION
**Pull request by tbirdso**
Suggesting comment updates for `itk::Spectra1DImageFilter` and `itk::Spectra1DSupportWindowImageFilter` adding a few details on procedure and filter output.

### CODE REVIEW

File: include/itkSpectra1DImageFilter.hxx [hunk 1]
```diff
@@ -144,10 +149,10 @@ Spectra1DImageFilter<TInputImage, TSupportWindowImage, TOutputImage>::ComputeSpe
   const double overlap = 0.5;
   IndexType    segmentIndex(lineIndex);
   const double spectralScale = 1.0 / (fftSize * fftSize);
-  for (unsigned int segment = 0; segment < 3; ++segment)
+  for (unsigned int segment = 0; segment < 3; ++segment) // Why 3?
```
Comment (tbirdso) [inline include/itkSpectra1DImageFilter.hxx:152]: @thewtex @dzenanz Could you please clarify how segments are being used here? Why are we using 3?

File: include/itkSpectra1DImageFilter.hxx [hunk 2]
```diff
@@ -164,10 +169,12 @@ Spectra1DImageFilter<TInputImage, TSupportWindowImage, TOutputImage>::ComputeSpe
     spectraVectorIt = perThreadData.SpectraVector.begin();
     // drop DC component
     ++complexVectorConstIt;
+
+    // Each spectral component = (Re^2 + Im^2) / 3 / (fftSize)^2
     for (size_t freq = 0; freq < highFreq; ++freq)
     {
       spectraVectorIt[freq] +=
-        std::real(*complexVectorConstIt * std::conj(*complexVectorConstIt)) / 3.0 * spectralScale;
+        std::real(*complexVectorConstIt * std::conj(*complexVectorConstIt)) / 3.0 * spectralScale; // Why 3?
```
Comment (tbirdso) [inline include/itkSpectra1DImageFilter.hxx:177]: @thewtex @dzenanz Same question - why 3.0?

File: include/itkSpectra1DImageFilter.hxx [hunk 3]
```diff
@@ -144,10 +149,10 @@ Spectra1DImageFilter<TInputImage, TSupportWindowImage, TOutputImage>::ComputeSpe
   const double overlap = 0.5;
   IndexType    segmentIndex(lineIndex);
   const double spectralScale = 1.0 / (fftSize * fftSize);
-  for (unsigned int segment = 0; segment < 3; ++segment)
+  for (unsigned int segment = 0; segment < 3; ++segment) // Why 3?
   {
-    segmentIndex[0] =
-      static_cast<IndexValueType>(lineIndex[0] + segment * perThreadData.LineImageRegionSize[0] * overlap / 3.0);
+    segmentIndex[0] = static_cast<IndexValueType>(lineIndex[0] + segment * perThreadData.LineImageRegionSize[0] *
+                                                                   overlap / 3.0); // Why 3?
```
Comment (tbirdso) [inline include/itkSpectra1DImageFilter.hxx:155]: @thewtex @dzenanz Why 3? Number of segments?

**Comment by thewtex** [comment]
@tbirdso 3 because we are using Welch's Method:

https://en.wikipedia.org/wiki/Welch%27s_method

with 50% overlap on segments that are half the total length.

**Comment by tbirdso** [comment]
Thanks @thewtex, I've updated comments accordingly.

**Review by dzenanz — APPROVED** [review]
Looks good to me. I don't know the answer to your question about 3 segments.

File: include/itkSpectra1DImageFilter.hxx [hunk 4]
```diff
@@ -144,6 +149,7 @@ Spectra1DImageFilter<TInputImage, TSupportWindowImage, TOutputImage>::ComputeSpe
   const double overlap = 0.5;
   IndexType    segmentIndex(lineIndex);
   const double spectralScale = 1.0 / (fftSize * fftSize);
+  // 3 segments for Welsh's method with 50% overlap (D=M/2)
```
Comment (dzenanz) [inline include/itkSpectra1DImageFilter.hxx:152]: Welch's, not Welsh's.

Comment (tbirdso) [reply include/itkSpectra1DImageFilter.hxx]: Whoops, good catch. Amended.

**Review by dzenanz — APPROVED** [review]

**Review by thewtex — APPROVED** [review]
:book: :rocket:

**Pull request merged**